### PR TITLE
refactor(bigquery): add table name to filter logadmin query

### PR DESCRIPTION
**Pull request by mabdh**
- add table name to filter logadmin query
- collect log per table instead of building a global hash map

### CODE REVIEW

**Comment by mabdh** [comment]
Need to increase coverage to `bigquery/auditlog`

File: plugins/extractors/bigquery/bigquery.go [hunk 1]
```diff
@@ -181,10 +174,20 @@ func (e *Extractor) extractTable(ctx context.Context, ds *bigquery.Dataset, emit
 
 // Build the bigquery table metadata
 func (e *Extractor) buildTable(ctx context.Context, t *bigquery.Table, md *bigquery.TableMetadata) *assetsv1beta1.Table {
+	var tableStats *auditlog.TableStats
+	if e.config.IsCollectTableUsage {
+		// Fetch and extract logs first to build a map
+		var errL error
+		tableStats, errL = e.galClient.Collect(ctx, t.TableID)
```
Comment (StewartJingga) [inline plugins/extractors/bigquery/bigquery.go:181]: CMIIW, so now we will fetch table usage once for each table, which will create more requests but we won't fetch unused data/rows? @mabdh

Comment (mabdh) [reply plugins/extractors/bigquery/bigquery.go:181]: @StewartJingga yes correct

**Review by StewartJingga — APPROVED** [review]
Code's LGTM

**Pull request merged**